runtime/permissions.rs

```diff
@@ -545,18 +546,31 @@ impl Permissions {
     )
   }
 
-  pub fn check_net(&self, hostname: &str, port: u16) -> Result<(), AnyError> {
-    self.query_net(hostname, Some(port)).check(
-      &format!("network access to \"{}:{}\"", hostname, port),
+  pub fn check_net<T: AsRef<str>>(
+    &self,
+    host: &(T, Option<u16>),
```

Comment:
Maybe `host: (&str, Option<u16>)` or split `host` and `port`? I think @magurotuna had similar problem in his DNS PR

Reply:
`(&str, Option<u16>)` doesn't work because it's really hard to pass a `String` when a `&str` is expected as part of a tuple. Splitting `host` and `port` avoids the issue, but it's inconsistent and not applicable with the other APIs where this is wrapped in an `Option`.

Reply:
Thanks for explanation, that sounds reasonable to me, let's keep as is